dht.go

```diff
@@ -303,6 +303,10 @@ func (dht *IpfsDHT) findPeerSingle(ctx context.Context, p peer.ID, id peer.ID) (
 	resp, err := dht.sendRequest(ctx, p, pmes)
 	switch err {
 	case nil:
+		// reset the timer for the k-bucket we just searched in ONLY if there was no error
+		// so that we can retry during the next bootstrap
+		bucket := dht.routingTable.BucketForPeer(id)
+		bucket.ResetLastQueriedAt(time.Now())
```

Comment:
Ah, I'm sorry, I think I got this wrong. I think we have to update buckets based on the _key_ we're querying (and we can do it whether or not the query succeeds, really), not based on whether or not we've contacted a peer in that bucket. Otherwise, we won't fill that bucket, we'll just add a single node.

We'll probably need to do this in _several_ places.

Reply:
@Stebalien Ah, I see exactly what you mean. Any lookup for a _key_ in a bucket will result in attempts to add _peers_ closer to that _key_ to the bucket and thats is essentially a _lookup_/_query_ on the bucket.